src/librbd/cache/pwl/ssd/WriteLog.cc

```diff
@@ -124,18 +124,33 @@ bool WriteLog<I>::initialize_pool(Context *on_finish,
                                nullptr, nullptr, nullptr);
     int r = bdev->open(this->m_log_pool_name);
     if (r < 0) {
+      lderr(m_image_ctx.cct) << "fail to open block device." << dendl;
       delete bdev;
-      on_finish->complete(-1);
+      on_finish->complete(r);
+      return false;
+    }
+    if (this->m_log_pool_size != bdev->get_size()) {
```

Comment:
I would check for `this->m_log_pool_size % MIN_WRITE_ALLOC_SSD_SIZE == 0` and then for
`this->m_log_pool_size == bdev->get_size()`.